presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/ElasticsearchMetadata.java

```diff
@@ -79,6 +95,16 @@ public ElasticsearchMetadata(TypeManager typeManager, ElasticsearchClient client
         this.client = requireNonNull(client, "client is null");
         requireNonNull(config, "config is null");
         this.schemaName = config.getDefaultSchema();
+
+        Type jsonType = typeManager.getType(new TypeSignature(StandardTypes.JSON));
+        queryResultColumnMetadata = ColumnMetadata.builder()
+                .setName("result")
+                .setType(jsonType)
```

Comment:
I was thinking about the relative advantages of using `JSON` vs `VARCHAR` for the column type.  `JSON` is more expressive (one might even argue correctness), but without knowing how a client might use the result, `VARCHAR` might be more pragmatic.  I think it's conceivable one might want to parse and process the results of an aggregation client side, so they might end up serializing the result and reading the resulting string.  The responses can get pretty big and the cost of parsing a big JSON document might not be insignificant.  On the other hand, if a client wanted to join the aggregation result with some other table, they'd only need to call `json_parse(result)`.

Reply:
I started with VARCHAR and then ended up changing it to JSON. In practice, I expect almost all usages to try to do something with the result as a JSON object. Parsing the result client side is a very roundabout way of interacting with ES. You might as well talk to it directly.

If you need the text, just use `json_format`. It's actually free under the covers, as the internal representation of the JSON type is the json string.